src/resources/ResourceSnapshots/ResourceSnapshots.ts

```diff
@@ -25,6 +26,26 @@ export default class ResourceSnapshots extends Resource {
         return await fetch(url, {method: 'get'});
     }
 
+    createFromFile(file: File, options: CreateFromFileOptions) {
+        const computedOptions = {developerNotes: options.developerNotes, snapshotFileType: undefined};
+
+        if (file.type === 'application/zip') {
+            computedOptions.snapshotFileType = 'ZIP';
+        } else if (file.type === 'application/json') {
+            computedOptions.snapshotFileType = 'JSON';
+        } else {
+            throw new Error('The uploaded file must be either a ZIP or a JSON file.');
```

Comment:
Have you tested it? When linked to the admin, does the toast show up?

Reply:
No, I will catch it in the Admin UI